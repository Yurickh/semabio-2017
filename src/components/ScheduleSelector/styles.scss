@import 'ambience/color';
@import 'ambience/responsive';
@import 'base/responsive-prop';

.schedule-selector {
  & > .week-picker {
    display: flex;
    justify-content: center;
  }

  & > .week-picker > .day {
    box-sizing: border-box;
    @include responsive-prop(padding-top, 14, 14);
    @include responsive-prop(padding-bottom, 14, 14);
    @include responsive-prop(padding-left, 34, 34);
    @include responsive-prop(padding-right, 34, 34);

    @include small-only {
      @include responsive-prop(padding-top, 7, 7);
      @include responsive-prop(padding-bottom, 7, 7);
      @include responsive-prop(padding-left, 10, 10);
      @include responsive-prop(padding-right, 10, 10);
    }

    cursor: pointer;

    color: $color-forest;
  }

  & > .week-picker > .day.-selected {
    @include responsive-prop(padding-top, 12, 12);
    @include responsive-prop(padding-bottom, 12, 12);
    @include responsive-prop(padding-left, 32, 32);
    @include responsive-prop(padding-right, 32, 32);

    border: $color-forest 2px solid;
    border-radius: 24px;

    @include small-only {
      @include responsive-prop(padding-top, 5, 5);
      @include responsive-prop(padding-bottom, 5, 5);
      @include responsive-prop(padding-left, 8, 8);
      @include responsive-prop(padding-right, 8, 8);
    }
  }
}
