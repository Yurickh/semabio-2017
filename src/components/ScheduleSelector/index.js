import React, { Component } from 'react'

import ScheduleTable from '../ScheduleTable'

import './styles.css'

class ScheduleSelector extends Component {
	state = {
		selected: 'SEG',
	}

	choose = (day) => {
		this.setState({
			selected: day
		})
	}

	render() {
		return (
			<main className="schedule-selector">
				<nav className="week-picker">
					{
						['SEG', 'TER', 'QUA', 'QUI', 'SEX', 'SAB'].map(day => {
							const selected = day === this.state.selected
							return (
								<div
									key={day}
									className={`day ${selected ? '-selected' : ''}`}
									onClick={() => this.choose(day)}
								>
									{day}
								</div>
							)
						})
					}
				</nav>

				<ScheduleTable selected={this.state.selected} />
			</main>
		)
	}
}

export default ScheduleSelector
