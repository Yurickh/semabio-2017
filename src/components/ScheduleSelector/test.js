import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import ScheduleSelector from '../ScheduleSelector'

describe('ScheduleSelector', () => {
  it('should render correctly', () => {
    const tree = shallow(<ScheduleSelector />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
