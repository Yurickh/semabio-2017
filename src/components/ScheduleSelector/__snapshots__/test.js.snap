// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleSelector should render correctly 1`] = `
<main
  className="schedule-selector"
>
  <nav
    className="week-picker"
  >
    <div
      className="day -selected"
      onClick={[Function]}
    >
      SEG
    </div>
    <div
      className="day "
      onClick={[Function]}
    >
      TER
    </div>
    <div
      className="day "
      onClick={[Function]}
    >
      QUA
    </div>
    <div
      className="day "
      onClick={[Function]}
    >
      QUI
    </div>
    <div
      className="day "
      onClick={[Function]}
    >
      SEX
    </div>
    <div
      className="day "
      onClick={[Function]}
    >
      SAB
    </div>
  </nav>
  <ScheduleTable
    selected="SEG"
  />
</main>
`;
